lib/parser.js

```diff
@@ -341,7 +341,14 @@ module.exports = function (args, opts, y18n) {
       var configPath = argv[configKey] || configLookup[configKey]
       if (configPath) {
         try {
-          var config = require(path.resolve(process.cwd(), configPath))
+          var config = null
+          var resolvedConfigPath = path.resolve(process.cwd(), configPath)
+
+          if (typeof flags.configs[configKey] === 'function') {
+            config = flags.configs[configKey](resolvedConfigPath)
+          } else {
+            config = require(resolvedConfigPath)
```

Comment:
Per #312, should we aspire to remove the dynamic `require()` here? Maybe use [`jsonfile`](https://www.npmjs.com/package/jsonfile) instead or check file extension first?

Or would you rather wait til 4.x to change this? I guess changing it could certainly cause backwards compatibility problems.